sw/host/tests/usbdev/usbdev_stream/stream_test.cc

```diff
@@ -0,0 +1,273 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+// USB streaming data test
+//
+// Linux host-side application that receives a stream of LFSR-generated data
+// from the USB device, checks the received bytestream and then XORs it with a
+// host-side LFSR-generated byte stream to transmit back to the device.
+//
+// By default the streaming test expects a number of USB serial connections to
+// the target device, one port per endpoint:
+//
+//   /dev/ttyUSB0 - supplies and receives LFSR-generated byte stream for one/
+//                  the only endpoint
+//   /dev/ttyUSB1 - a secondary stream
+//   /dev/ttyUSB..
+//
+// Note that the mapping from device endpoints to USB port number is not
+// guaranteed, and  when multiple streams are used, it is _not_ necessarily the
+// case that ascending streams/endpoints in usbdev_stream_test are mapped to
+// a contiguous range of ascending ttyUSBi port names.
+//
+// Either or both of the initial input port and the initial output port may be
+// overridden using command line parameters.
+//
+// Usage:
+//   stream [-n<streams>][-v<bool>][-c<bool>][-r<bool>][-s<bool>]
+//          [<input port>[ <output port>]]
```

Comment:
It is consistent and predictable on my laptop, but I could believe that the CI systems are different. Hopefully it will be the case that the port names will always be the same. They appear as /dev/ttyUSBn, so will depend upon the number of other 'serial over USB' connections.

It would be good to get this test into CI once it has been merged, but later some of the code will need to use libusb which allows us to identify our device specifically. If we can't rely upon the port names being predictable, then maybe it will just be easier to make all of the streams operate through libusb.

Reply:
We can instrument CI so that USB devices get a predictable symlink.

We already do that for the OT USB device:
```
--symlink usb:18d1:503a:0=/dev/ttyUSB_OT_0
--symlink usb:18d1:503a:1=/dev/ttyUSB_OT_1
```
That reads: For the USB device with vid=18d1 pid=503a, interfaces 0 and 1, we create /dev/ttyUSB_OT_0 and /dev/ttyUSB_OT_1

If the VID, PID, or interfaces of the OT USB device have changed, we need to create new rules for the symlinks.